tests/common.py

```diff
@@ -984,18 +984,20 @@ def assert_setup_component(count, domain=None):
     async def mock_psc(hass, config_input, integration):
         """Mock the prepare_setup_component to capture config."""
         domain_input = integration.domain
-        res = await async_process_component_config(hass, config_input, integration)
+        res, exceptions = await async_pre_process_component_config(
+            hass, config_input, integration
+        )
```

Comment:
This is needed because this acts as a wrapper, Since we process the data in async_process_component_config we need to wrap this instead.